lib/chore/job.rb

```diff
@@ -60,6 +60,12 @@ def queue_options(opts = {})
             raise ArgumentError, "#{self.to_s}: backoff must accept a single argument"
           end
         end
+
+        if @chore_options.key?(:dedupe_lambda)
+          if !@chore_options[:dedupe_lambda].is_a?(Proc)
+            raise ArgumentError, "#{self.to_s}: dedupe_lambda must be a lambda"
```

Comment:
Strictly speaking lambdas are only a subset of Procs. I'm not sure that the difference will matter much here.

Reply:
yea, it seems like the only way to check if something is a lambda would be to do something like `@chore_options[:dedupe_lambda].respond_to? :call` which seems weirder. If they  pass a proc, it'll fail on the arity check anyways

Reply:
Procs respond to call too.

Reply:
The arity only fails when you have default arguments in the proc.  I would suggest keeping the error messaging the same -- `backoff must be a lambda or Proc`.   Having some options take in a lambda-only and others take in lambda or proc is a little confusing.